packages/backend/src/league/league_controller.ts

```diff
@@ -265,4 +268,62 @@ export class LeaguesController extends ControllerWrap<League> {
       }),
     );
   }
+
+  /**
+   * Generates match proposal
+   */
+  @Post("{leagueId}/generate-match-proposals")
+  public async generateMatchProposals(
+    @Path() leagueId: ID,
+    @Request() req: express.Request,
+  ): Promise<WithError<MatchProposal[]>> {
+    const id = new ObjectId(leagueId);
+
+    return this.withUser(req, async (currUser) => {
+      const res = await this.get(id);
+      if (!res.success) {
+        return res;
+      }
+
+      const league = res.data;
+      if (!hasId(league.ownerIds, currUser._id)) {
+        return this.notFound();
+      }
+
+      const matchProposals: MatchProposal[] = [];
+
+      // Group people by ability level
+      const peopleByAbilityLevel: Record<AbilityLevel, ObjectId[]> = {
+        beginner: [],
+        intermediate: [],
+        advanced: [],
+      };
+      for (const person of league.people as User[]) {
+        peopleByAbilityLevel[person.abilityLevel].push(person._id);
```

Comment:
`abilityLevel` does not appear on the person as it is dependent on the sport. (See `User.sports`)